From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@google.com>
Date: Mon, 22 Jun 2020 17:44:11 +0200
Subject: Revert "module: Make module_enable_ro() static again"

This reverts commit e6eff4376e2897c2e14b70d87bf7284cdb093830.

It causes a build-breakage on the network test configuration so revert
it for now to hopefully add back after -rc2 is merged.

Signed-off-by: Greg Kroah-Hartman <gregkh@google.com>
Change-Id: I866dcc31dc3d3eb68929708d28935c5e13a7fb3d
---
 include/linux/module.h | 6 ++++++
 kernel/module.c        | 3 +--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/include/linux/module.h b/include/linux/module.h
--- a/include/linux/module.h
+++ b/include/linux/module.h
@@ -866,6 +866,12 @@ extern int module_sysfs_initialized;
 
 #define __MODULE_STRING(x) __stringify(x)
 
+#ifdef CONFIG_STRICT_MODULE_RWX
+extern void module_enable_ro(const struct module *mod, bool after_init);
+#else
+static inline void module_enable_ro(const struct module *mod, bool after_init) { }
+#endif
+
 #ifdef CONFIG_GENERIC_BUG
 void module_bug_finalize(const Elf_Ehdr *, const Elf_Shdr *,
 			 struct module *);
diff --git a/kernel/module.c b/kernel/module.c
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -2000,7 +2000,7 @@ static void frob_writable_data(const struct module_layout *layout,
 		   (layout->size - layout->ro_after_init_size) >> PAGE_SHIFT);
 }
 
-static void module_enable_ro(const struct module *mod, bool after_init)
+void module_enable_ro(const struct module *mod, bool after_init)
 {
 	if (!rodata_enabled)
 		return;
@@ -2028,7 +2028,6 @@ static void module_enable_nx(const struct module *mod)
 
 #else /* !CONFIG_STRICT_MODULE_RWX */
 static void module_enable_nx(const struct module *mod) { }
-static void module_enable_ro(const struct module *mod, bool after_init) {}
 #endif /*  CONFIG_STRICT_MODULE_RWX */
 static void module_enable_x(const struct module *mod)
 {
